src/training_set.py

```diff
@@ -0,0 +1,294 @@
+        igmsp_survey = 'SDSS_DR7'
+    nsight = len(dlasurvey.sightlines)
+    keep = np.array([True]*nsight)
+    meta = Table(igmsp.idb.hdf[igmsp_survey+'/meta'].value)
+
+    # Avoid DLAs
+    dla_coord = dlasurvey.coord
+    sl_coord = SkyCoord(ra=dlasurvey.sightlines['RA'], dec=dlasurvey.sightlines['DEC'])
+    idx, d2d, d3d = match_coordinates_sky(sl_coord, dla_coord, nthneighbor=1)
+    clear = d2d > 1*u.arcsec
+    keep = keep & clear
+
+    # BAL
+    if flg_bal is not None:
+        gd_bal = dlasurvey.sightlines['FLG_BAL'] <= flg_bal
+        keep = keep & gd_bal
+
+    # S/N
+    if s2n > 0.:
+        gd_s2n = dlasurvey.sightlines['S2N'] > s2n
+        keep = keep & gd_s2n
+
+    # Cut on DX
+    if DX > 0.:
+        gd_DX = dlasurvey.sightlines['DX'] > DX
+        keep = keep & gd_DX
+
+    # igmsp
+    qso_coord = SkyCoord(ra=meta['RA'], dec=meta['DEC'], unit='deg')
+    idxq, d2dq, d3dq = match_coordinates_sky(sl_coord, qso_coord, nthneighbor=1)
+    in_igmsp = d2dq < 1*u.arcsec
+    keep = keep & in_igmsp
+
+    # Assess
+    final = dlasurvey.sightlines[keep]
+    sdict = {}
+    sdict['n'] = len(final)
+    print("We have {:d} sightlines for analysis".format(sdict['n']))
+
+    def qck_stats(idict, tbl, istr, key):
+        idict[istr+'min'] = np.min(tbl[key])
+        idict[istr+'max'] = np.max(tbl[key])
+        idict[istr+'median'] = np.median(tbl[key])
+    qck_stats(sdict, final, 'z', 'ZEM')
+    qck_stats(sdict, final, 'i', 'MAG')
+
+    print("Min z = {:g}, Median z = {:g}, Max z = {:g}".format(sdict['zmin'], sdict['zmedian'], sdict['zmax']))
+
+    # Return
+    return final, sdict
+
+
+def init_fNHI():
+    """ Generate the interpolator for log NHI
+
+    Returns
+    -------
+    fNHI : scipy.interpolate.interp1d function
+    """
+    from pyigm.fN.fnmodel import FNModel
+    # f(N)
+    fN_model = FNModel.default_model()
+    # Integrate on NHI
+    lX, cum_lX, lX_NHI = fN_model.calculate_lox(fN_model.zpivot,
+                                                20.3, NHI_max=22.5, cumul=True)
+    # Interpolator
+    cum_lX /= cum_lX[-1] # Normalize
+    fNHI = interpolate.interp1d(cum_lX, lX_NHI,
+                                bounds_error=False,fill_value=lX_NHI[0])
+    return fNHI
+
+
+def insert_dlas(spec, zem, fNHI=None, rstate=None):
+    """
+    Parameters
+    ----------
+    spec
+    fNHI
+    rstate
+
+    Returns
+    -------
+    final_spec : XSpectrum1D
+    dlas : list
+      List of DLAs inserted
+
+    """
+    from pyigm.fN import dla as pyi_fd
+    from pyigm.abssys.dla import DLASystem
+    from pyigm.abssys.utils import hi_model
+
+    # Init
+    if rstate is None:
+        rstate = np.random.RandomState()
+    if fNHI is None:
+        fNHI = init_fNHI()
+
+    # Allowed redshift placement
+    ## Cut on zem and 920A rest-frame
+    zlya = spec.wavelength.value/1215.67 - 1
+    dz = np.roll(zlya,-1)-zlya
+    dz[-1] = dz[-2]
+    gdz = (zlya < zem) & (spec.wavelength > 910.*u.AA*(1+zem))
```

Comment:
Probably not, but which way did you mean! ;)

No really, were you thinking to go to lower wavelengths (where LLS
cause trouble) or larger?  

@davidparks21 -- can you remind me of which rest wavelength begins
your analysis?  I seem to recall 920A.